keps/sig-apps/4212-declarative-node-maintenance/kep.yaml

```
@@ -0,0 +1,52 @@
+title: Declarative Node Maintenance
+kep-number: 4212
+authors:
+  - "@atiratree"
+owning-sig: sig-apps
+participating-sigs:
+  - sig-apps
+  - sig-autoscaling
+  - sig-cli
+  - sig-node
+  - sig-scheduling
+status: provisional
+creation-date: 2023-09-15
+reviewers:
+  - "@sftim"
+  - "@fabiand"
+  - "@mmerkes"
+  - "@rptaylor"
+  - "@pwschuurman"
+  - "@dchen1107"
+  - "@soltysh"
+approvers:
+  - "@dchen1107"
+  - "@soltysh"
+
+# The target maturity stage in the current dev cycle for this KEP.
+stage: alpha
+
+# The most recent milestone for which work toward delivery of this KEP has been
+# done. This can be the current (upcoming) milestone, if it is being actively
+# worked on.
+latest-milestone: "v1.31"
+
+# The milestone at which this feature was, or is targeted to be, at each stage.
+milestone:
+  alpha: "v1.31"
+
+# The following PRR answers are required at alpha release
+# List the feature gate name and the components for which it must be enabled
+feature-gates:
+  - name: DeclarativeNodeMaintenance
+    components:
+      - kube-apiserver
+      - kube-controller-manager
+  - name: NodeMaintenanceDeployment
```

How about:
```suggestion
  - name: NodeMaintenanceAwareDeployments
```

Feature gate names [aim to] describe the behavior that will exist once this is GA.

---

This looks better - will rename